config/initializers/secure_headers.rb

```diff
@@ -14,11 +12,17 @@
   config.x_content_type_options = "nosniff"
 
   # http://msdn.microsoft.com/en-us/library/dd565647%28v=vs.85%29.aspx
-  config.x_xss_protection = { :value => 1 }
+  config.x_xss_protection = "1; mode=block"
 
   # https://w3c.github.io/webappsec/specs/content-security-policy/
-  config.csp = false
+  config.csp = {
+    # "meta" values. these will shaped the header, but the values are not included in the header.
+    report_only:  true,
+
+    # directive values: these values will directly translate into source directives
+    default_src: %w(https: 'self'),
+  }
 
   # https://www.nwebsec.com/HttpHeaders/SecurityHeaders/XDownloadOptions
-  config.x_download_options = false
+  config.x_download_options = nil
```

Comment:
Potentially, although this is an odd one...

(from https://www.nwebsec.com/HttpHeaders/SecurityHeaders/XDownloadOptions)

> The X-Download-Options is specific to IE 8, and is related to how IE 8 handles downloaded HTML files. Turns out if you download an HTML file from a web page and chooses to "Open" it in IE, it will execute in the context of the web site. That means that any scripts in that file will also execute with the origin of the web site.

So I'm tempted to set it to "noopen" which I'm taking to mean protect IE8 users from <strike>themselves</strike> accidentally running scripts locally. Although sending extra bytes to everyone else isn't massively useful.